lib/components/MenuRenderer/MenuRenderer.tsx

```diff
@@ -166,16 +166,20 @@ export const MenuRenderer = ({
   );
 
   useEffect(() => {
-    if (open) {
-      hasOpened.current = true;
+    if (lastOpen.current === open) {
+      return;
+    }
 
+    if (open) {
       if (typeof onOpen === 'function') {
```

Comment:
Any reason not to collapse these two `if`s into one?

Reply:
Yep, because it `onOpen` is not a function we don't want to trigger the else clause.

Reply:
But wouldn't `open && typeof onOpen === 'function'` achieve that, or am I missing something?